### DISCUSSION TITLE
How to detect the X-Rite ColorChecker Digital SG?

### DISCUSSION THREAD

**Discussion by gimlidc** [colour-science/colour-checker-detection#40] [Q&A] [answered]
I have inspired in your develop branch for X-Rite Color Checker DIGITAL. But I have this issue (see the right most column, which si somehow stretched from original image):

![Screenshot 2021-11-04 at 18 42 03](https://user-images.githubusercontent.com/1388882/140392112-38d569ae-a8c5-4c30-b495-efb483cfd549.png)

Here is the original image
![detection](https://user-images.githubusercontent.com/1388882/140391884-e3d9a901-1660-4108-ad3d-f60ff6449c69.jpg)

What is the proper configuration necessary for this file? Or what am I doing wrong?

**Comment by KelSolaar** [comment, marked as answer]
Hi @gimlidc,

I think it is an unfortunate case of having the checker right next to the border, thus the clustering is producing weird results, I just extend the image in the top-right by ~10% and it works fine then:

![image](https://user-images.githubusercontent.com/99779/140624370-b9a90a86-b8ec-4e18-8809-d56ec643d5a9.png)
![image](https://user-images.githubusercontent.com/99779/140624375-6e5c7921-50db-4c8f-ad88-9e92638c5bdb.png)

By the way, there is already an attribute that stores settings appropriate for the SG, i.e. `colour_checker_detection.SETTINGS_SEGMENTATION_COLORCHECKER_SG`:

```python
detect_colour_checkers_segmentation(
        image, additional_data=True, **SETTINGS_SEGMENTATION_COLORCHECKER_SG)
```
